### ISSUE TITLE
Feature Request: Preserve header location when organizing imports

### ISSUE THREAD

**Issue by AHelper** [buehler/typescript-hero#216]
#### System information

- OS: Windows 7
- TypeScript Hero version: 1.0.0
- VSCode version: 1.13.1

#### Feature: Preserve header location when organizing imports

Currently, if I have a file such as:
```typescript
// (c) <year> <name>
// Legal text...

import { Foo } from "foo";
import { Bar } from "bar";

...
```

If I organize imports, it will place the organized block of imports above the copyright header (although after "use strict"), which I have to keep moving back down to where it was (Same when adding a new import, it will get placed at the top of the file, but may want to split that out to a separate issue). 

It would be nice if imports were placed better.  Possibly placing the sorted block of imports at the location of the first import in the file instead would work for general cases where imports are already in the right location, but I don't know about use cases where imports need to get moved up.

(Seems like this was brought up before in https://github.com/buehler/typescript-hero/issues/73, but a new issue wasn't created)

**Comment by buehler** [comment]
Yea, the problem is, that I parse the files with the official `typescript` package.
To find comments, I'd need to "parse" the file with regex which would kill the performance of the extension. 

What I could do though (for simplicity and the sake of performance):
- Check (either all or the first few, I've not decided yet) lines of the actual document and check it with string.startWith
- If the following comment is found: `// TSH:Imports` then use the line below that comment as starting point for imports

What I'm not gonna do is parse the whole file with regex since that would kill the whole idea of the typescript parser.

What do you think about that purposal @AHelper ?

**Comment by AHelper** [comment]
Hmm, in my case of working inside an existing large codebase, adding in markers won't be very practical...  I can think of a workaround with a string.startsWith, if the string to search for can be a set in settings.json, then maybe searching for the last line in the header comment block would get it near the right spot, but doesn't seem quite right.

Looking at what TS does for quick fix on imports, it seems to find the first import statement or the first occurrence of code (non-comment, row & column, not just by line) and adds the import, padding with a blank line after if its the first import added.  Maybe something similar? I haven't looked at parsing with TS thoroughly nor how this extension uses it, but something along the lines of searching the AST with similar rules, get the position of the node and use that as the insertion point for imports?

**Comment by buehler** [comment]
Actually this may be an option, to use the first import as "start of imports". If there are no imports at all, start at the top of the file, otherwise use the starting point of the first import that is there in the file.

The parser goes through the files, parses them to an AST with the help of the `typescript` compiler. Afterwards, I parse the AST (recursive) to generate a set of usable classes and interfaces.

**Comment by buehler** [comment]
Closed by #296
